client/adventureMap/AdventureMapShortcuts.cpp

```diff
@@ -457,6 +463,72 @@ void AdventureMapShortcuts::zoom( int distance)
 	owner.hotkeyZoom(distance, false);
 }
 
+void AdventureMapShortcuts::search(bool next)
+{
+	// get all relevant objects
+	std::vector<ObjectInstanceID> visitableObjInstances;
+	int3 mapSizes = LOCPLINT->cb->getMapSize();
+	for(int x = 0; x < mapSizes.x; x++)
+		for(int y = 0; y < mapSizes.y; y++)
+			for(int z = 0; z < mapSizes.z; z++)
+				for(auto & obj : LOCPLINT->cb->getVisitableObjs(int3(x, y, z), false))
+					if(obj->ID != MapObjectID::MONSTER && obj->ID != MapObjectID::EVENT && obj->ID != MapObjectID::HERO)
+						visitableObjInstances.push_back(obj->id);
```

Comment:
Sounds like rather overcomplicated solution.
Perhaps add method `getAllVisitableObjects` to `CGameInfoCallback` that will do the same via a single pass over map->objects?